test.sh

```diff
@@ -1,10 +1,25 @@
+#!/bin/sh
+
+version="0.0.1"
+what_arch=`uname -m`
+
+if [ ${what_arch} == "x86_64" ]
+then
+    arch="amd64"
+elif [ ${what_arch} == "arm64"]
+then
+    arch=${what_arch}
+fi
+
+os=$(uname -s | tr '[:upper:]' '[:lower:]')
+
 go build
 
-dest=~/.terraform.d/plugins/terraform.local/local/go/0.0.1/darwin_arm64/terraform-provider-go_v0.0.1
+dest=~/.terraform.d/plugins/terraform.local/local/go/${version}/${os}_${arch}/terraform-provider-go_v${version}
```

Comment:
This is better than the original hard-coded value, but I think using [developer overrides](https://opentofu.org/docs/cli/config/config-file/#development-overrides-for-provider-developers) and creating a `~/.tofurc` file is probably a more appropriate solution.

1. `go install ./...`
2. Add `dev_overrides {…}`:

    ```hcl
    provider_installation {
      dev_overrides {
        "opentofu/go" = "/your/go/bin/dir"
      }
    }
    ```